definitions/npm/rxjs_v6.x.x/flow_v0.34.x-/rxjs_v6.x.x.js

```diff
@@ -1423,119 +1238,146 @@ declare module "rxjs/operators" {
-
-    withLatestFrom<+T, A, B, C, D, E>(
-      a: rxjs$Observable<A>,
-      b: rxjs$Observable<B>,
-      c: rxjs$Observable<C>,
-      d: rxjs$Observable<D>,
-      resultSelector: (t: T, a: A, b: B, c: C, d: D) => E
-    ): rxjs$Observable<T> => rxjs$Observable<E>;
-
-    withLatestFrom<+T, A, B, C, D>(
-      a: rxjs$Observable<A>,
-      b: rxjs$Observable<B>,
-      c: rxjs$Observable<C>,
-      resultSelector: (t: T, a: A, b: B, c: C) => D
-    ): rxjs$Observable<T> => rxjs$Observable<D>;
+    withLatestFrom: rxjs$Operators$withLatestFrom;
 
-    withLatestFrom<+T, A, B, C>(
-      a: rxjs$Observable<A>,
-      b: rxjs$Observable<B>,
-      resultSelector: (t: T, a: A, b: B) => C
-    ): rxjs$Observable<T> => rxjs$Observable<C>;
+    // deprecated in favor of static zip
+    // see: https://rxjs-dev.firebaseapp.com/api/operators/zip
+    zip: rxjs$Operators$zip;
 
-    withLatestFrom<+T, A, B>(
-      a: rxjs$Observable<A>,
-      resultSelector: (t: T, a: A) => B
-    ): rxjs$Observable<T> => rxjs$Observable<B>;
-
-    withLatestFrom<+T, A, B, C, D, E, F, G>(
-      a: rxjs$Observable<A>,
-      b: rxjs$Observable<B>,
-      c: rxjs$Observable<C>,
-      d: rxjs$Observable<D>,
-      e: rxjs$Observable<E>,
-      f: rxjs$Observable<F>,
-      g: rxjs$Observable<G>,
-      _: void
-    ): rxjs$Observable<T> => rxjs$Observable<[T, A, B, C, D, E, E, F, G]>;
-
-    withLatestFrom<+T, A, B, C, D, E, F>(
-      a: rxjs$Observable<A>,
-      b: rxjs$Observable<B>,
-      c: rxjs$Observable<C>,
-      d: rxjs$Observable<D>,
-      e: rxjs$Observable<E>,
-      f: rxjs$Observable<F>,
-      _: void
-    ): rxjs$Observable<T> => rxjs$Observable<[T, A, B, C, D, E, F]>;
-
-    withLatestFrom<+T, A, B, C, D, E>(
-      a: rxjs$Observable<A>,
-      b: rxjs$Observable<B>,
-      c: rxjs$Observable<C>,
-      d: rxjs$Observable<D>,
-      e: rxjs$Observable<E>,
-      _: void
-    ): rxjs$Observable<T> => rxjs$Observable<[T, A, B, C, D, E]>;
-
-    withLatestFrom<+T, A, B, C, D>(
-      a: rxjs$Observable<A>,
-      b: rxjs$Observable<B>,
-      c: rxjs$Observable<C>,
-      d: rxjs$Observable<D>,
-      _: void
-    ): rxjs$Observable<T> => rxjs$Observable<[T, A, B, C, D]>;
-
-    withLatestFrom<+T, A, B, C>(
-      a: rxjs$Observable<A>,
-      b: rxjs$Observable<B>,
-      c: rxjs$Observable<C>,
-      _: void
-    ): rxjs$Observable<T> => rxjs$Observable<[T, A, B, C]>;
-
-    withLatestFrom<+T, A, B>(
-      a: rxjs$Observable<A>,
-      b: rxjs$Observable<B>,
-      _: void
-    ): rxjs$Observable<T> => rxjs$Observable<[T, A, B]>;
-
-    withLatestFrom<+T, A>(a: rxjs$Observable<A>, _: void): rxjs$Observable<T> => rxjs$Observable<[T, A]>;
+    // deprecated in favor of static combineLatest
+    // see: https://rxjs-dev.firebaseapp.com/api/operators/combineLatest
+    combineLatest: rxjs$Operators$combineLatest;
 
     refCount<T>(): rxjs$ConnectableObservable<T> => rxjs$Observable<T>;
   };
 }
 
+interface rxjs$Operators$withLatestFrom {
```

Comment:
interfaces have been known to slow down libdefs. Can you confirm that it's necessary for this to be an `interface` and not something else?

Reply:
Is it enough to rewrite `interface` with `type` ? Also works after applying [my rewriting patch](https://gist.github.com/berlysia/b07685aca114ab3ea8164cb8f43b14b8) .

I've already tried some patterns of implementation but every of those were not work.

Tried & failed patterns:
- in the manner of overloading, same as static [forkJoin](https://github.com/flow-typed/flow-typed/blob/7deafd3517fd8934f288f1f14393c748c5588825/definitions/npm/rxjs_v6.x.x/flow_v0.34.x-/rxjs_v6.x.x.js#L446)/[zip](https://github.com/flow-typed/flow-typed/blob/7deafd3517fd8934f288f1f14393c748c5588825/definitions/npm/rxjs_v6.x.x/flow_v0.34.x-/rxjs_v6.x.x.js#L719) operator
    - reordered upside down(decreasing in number of arguments)
    - pairwised with presence / absence of resultSelector (also inc/dec in number of arguments)

- in the manner of intersection, same as static [combineLatest](https://github.com/flow-typed/flow-typed/blob/7deafd3517fd8934f288f1f14393c748c5588825/definitions/npm/rxjs_v6.x.x/flow_v0.34.x-/rxjs_v6.x.x.js#L319) operator
    - (same as above)